sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -104,6 +117,123 @@ class SentryOkHttpInterceptor(
+//            synthetic = true
+        }
+        val exception = SentryHttpClientError("Event was captured because the request status code was ${response.code}")
+        // TODO: okhttp thread uses thread pool so its always the same stack trace (okhttp frames only), stacktrace is set as snapshot=true, is that ok?
+        val mechanismException = ExceptionMechanismException(mechanism, exception, Thread.currentThread(), true)
+        val event = SentryEvent(mechanismException)
+
+        // TODO: Do we need a message if the request and response already have the info?
+        val sentryMessage = Message().apply {
+            message = "HTTP url: %s status: %s"
+            params = listOf(requestUrl, response.code.toString())
+        }
+
+        val hint = Hint()
+        hint.set("request", request)
+        hint.set("response", response)
+
+        // TODO: remove after fields indexed
+        val tags = mutableMapOf<String, String>()
+        tags["status_code"] = response.code.toString()
+        tags["url"] = requestUrl
+
+        val unknownRequestFields = mutableMapOf<String, Any>()
+
+        val sentryRequest = io.sentry.protocol.Request().apply {
+            url = requestUrl
+            cookies = request.headers["Cookie"]
+            method = request.method
+            queryString = query
+            headers = getHeaders(request.headers)
+            fragment = urlFragment
+
+            request.body?.contentLength().ifHasValidLength {
+                // TODO: should be mapped in relay and added to the protocol
+                unknownRequestFields["body_size"] = it
+            }
+
+            unknown = unknownRequestFields.ifEmpty { null }
+        }
+
+        val sentryResponse = io.sentry.protocol.Response().apply {
+            cookies = response.headers["Cookie"]
+            headers = getHeaders(response.headers)
+            statusCode = response.code
+
+            response.body?.contentLength().ifHasValidLength {
+                bodySize = it
+            }
+        }
+
+        event.tags = tags
+        event.request = sentryRequest
+        event.contexts.setResponse(sentryResponse)
+        event.message = sentryMessage
+
+        hub.captureEvent(event, hint)
+    }
+
+    private fun containsStatusCode(statusCode: Int): Boolean {
+        for (item in failedRequestStatusCode) {
+            if (item.isInRange(statusCode)) {
+                return true
+            }
+        }
+        return false
+    }
+
+    private fun getHeaders(requestHeaders: Headers): MutableMap<String, String>? {
+        // TODO: should it be under isSendDefaultPii?
+        // Server already does data scrubbing
+        if (!hub.options.isSendDefaultPii) {
+            return null
+        }
+
+        val headers = mutableMapOf<String, String>()
+
+        for (i in 0 until requestHeaders.size) {
+            // TODO: should we remove sentry-trace and baggage from headers?
```

Comment:
@jjbayer since they are part of the headers, should we actually remove them on the client? should the server do it? or should we just keep them there?

Reply:
As discussed offline, let's send them to the server. If there is a collective decision that we do not want to see these headers in the product, we can drop them in Relay.